### PR TITLE
Added Memory Observer.

### PR DESCRIPTION
**Pull request by twevs**
Accessed via Debug > Show Memory Observer, provides some delta-over-time search functionality.

### CODE REVIEW

File: src/gui/widgets/memory_observer.cc [hunk 1]
```diff
@@ -0,0 +1,222 @@
+/***************************************************************************
+ *   Copyright (C) 2019 PCSX-Redux authors                                 *
```
Comment (nicolasnoble) [inline src/gui/widgets/memory_observer.cc:2]: 2022 :)

File: src/gui/widgets/memory_observer.cc [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#include "core/psxmem.h"
+#include "core/system.h"
+
+void PCSX::Widgets::MemoryObserver::draw(const char* title) {
+    if (!ImGui::Begin(title, &m_show)) {
+        ImGui::End();
+        return;
+    }
+
+    uint8_t* memData = g_emulator->m_psxMem->g_psxM;
```
Comment (nicolasnoble) [inline src/gui/widgets/memory_observer.cc:34]: Probably `const uint8_t*` here.

File: src/gui/widgets/memory_observer.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    static int getMemValue(uint32_t absoluteAddress, const uint8_t* memData, uint32_t memSize, uint32_t memBase, uint8_t stride);
+
+    ScanType m_scanType = ScanType::ExactValue;
+    ScanAlignment m_scanAlignment = ScanAlignment::OneByte;
+    std::vector<AddressValuePair> m_AddressValuePairs;
+    bool m_hex = false;
+    int m_value = 0;
+    bool m_showMemoryEditor = false;
+
+    MemoryEditor m_memoryEditor;
```
Comment (nicolasnoble) [inline src/gui/widgets/memory_observer.h:70]: We could avoid creating yet another memory editor, and simply use the `JumpToMemory` event to trigger the main memory editor to move, like this:

https://github.com/grumpycoders/pcsx-redux/blob/3b28a1d6be1423ed535d40364aa75dca5d4e0c3c/src/gui/widgets/assembly.cc#L299

File: src/gui/widgets/memory_observer.cc [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    uint32_t memSize = 8 * 1024 * 1024;
```
Comment (nicolasnoble) [inline src/gui/widgets/memory_observer.cc:35]: Might be worth doing something like

```c
const uint32_t memSize = 1024 * 1024 * (g_emulator->settings.get<PCSX::Emulator::Setting8MB>() 8 : 2)? 
```

File: src/gui/widgets/memory_observer.cc (hunk 1 above)
Comment (twevs) [reply src/gui/widgets/memory_observer.cc]: Ha, I had a variety of project files open as references. :)

**Comment by twevs** [comment]
Thank you for your feedback, I have pushed the changes.

**Review by nicolasnoble — DISMISSED** [review]

**Pull request merged**